hooks/config.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const { loadConfig } = require('../lib/config/index');
+const { normalizeConfig } = require('../lib/config/utils');
+
+exports.id = 'ti.config.js';
+exports.init = (logger, config, cli) => {
+	cli.on('build.config', {
+		post(data) {
+			const command = data.result[1];
+			const projectDirOption = command.options['project-dir'];
+			if (!projectDirOption) {
+				return;
+			}
+
+			const originalCallback = projectDirOption.callback;
+			projectDirOption.callback = function (projectDir) {
+				if (projectDir === '') {
+					// no option value was specified
+					// set project dir to current directory
+					projectDir = projectDirOption.default;
+				}
+				projectDir = path.resolve(projectDir);
+
+				const tiConfigPath = path.resolve(projectDir, 'ti.config.js');
+				if (!fs.existsSync(tiConfigPath) || fs.existsSync(path.join(projectDir, 'timodule.xml'))) {
+					return originalCallback.call(null, projectDir);
+				}
+
+				const tiConfig = loadConfig(projectDir);
+				const tiapp = normalizeConfig(tiConfig);
+				cli.argv.type = 'app';
+				cli.tiapp = tiapp;
+
+				// make sure the tiapp config is sane
+				const tiModulePath = require.resolve('node-titanium-sdk', { paths: [ cli.sdk.path ] });
+				const ti = require(tiModulePath);
```

Comment:
<!--
  0 failure: 
  1 warning:  hooks/config.js l...
  
  
  DangerID: danger-id-Danger;
  File: hooks/config.js;
  Line: 40;
-->

- :warning: hooks/config.js line 40 – Found non-literal argument in require (security/detect-non-literal-require)